cpuid causes test cases to fail on non-x86 platforms

This was first internally reported by @SimeonEhrig who encountered this on an ARM device. The offending line is here:

https://github.com/alpaka-group/alpaka/blob/261bdf70f359b3d97dfdfb3cc2bd39ec0472c8d1/include/alpaka/dev/cpu/SysInfo.hpp#L74

Error log:

```
[  3%] Building CXX object test/analysis/headerCheck/CMakeFiles/headerCheck.dir/src/acc/AccCpuOmp2Blocks.hpp.cpp.o

In file included from /home/simeon/project/alpaka/include/alpaka/dev/DevCpu.hpp:14,
                 from /home/simeon/project/alpaka/include/alpaka/acc/AccCpuOmp2Blocks.hpp:45,
                 from /home/simeon/project/alpaka/build/test/analysis/headerCheck/src/acc/AccCpuOmp2Blocks.hpp.cpp:1:
/home/simeon/project/alpaka/include/alpaka/dev/cpu/SysInfo.hpp: In function ‘void alpaka::cpu::detail::cpuid(uint32_t, uint32_t, uint32_t*)’:
/home/simeon/project/alpaka/include/alpaka/dev/cpu/SysInfo.hpp:74:51: error: unused parameter ‘level’ [-Werror=unused-parameter]
   74 |             inline auto cpuid(std::uint32_t const level, std::uint32_t const subfunction, std::uint32_t ex[4]) -> void
      |                               ~~~~~~~~~~~~~~~~~~~~^~~~~
/home/simeon/project/alpaka/include/alpaka/dev/cpu/SysInfo.hpp:74:78: error: unused parameter ‘subfunction’ [-Werror=unused-parameter]
   74 |             inline auto cpuid(std::uint32_t const level, std::uint32_t const subfunction, std::uint32_t ex[4]) -> void
      |                                                          ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
cc1plus: all warnings being treated as errors
gmake[2]: *** [test/analysis/headerCheck/CMakeFiles/headerCheck.dir/build.make:95: test/analysis/headerCheck/CMakeFiles/headerCheck.dir/src/acc/AccCpuOmp2Blocks.hpp.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2131: test/analysis/headerCheck/CMakeFiles/headerCheck.dir/all] Error 2
gmake: *** [Makefile:160: all] Error 2
```

--- a/include/alpaka/dev/cpu/SysInfo.hpp
+++ b/include/alpaka/dev/cpu/SysInfo.hpp
@@ -52,26 +52,26 @@ namespace alpaka
 #    if BOOST_COMP_GNUC || BOOST_COMP_CLANG || (!BOOST_COMP_MSVC_EMULATED && defined(__INTEL_COMPILER))               \
         || BOOST_COMP_PGI
 #        include <cpuid.h>
-            inline auto cpuid(std::uint32_t const level, std::uint32_t const subfunction, std::uint32_t ex[4]) -> void
+            inline auto cpuid(std::uint32_t level, std::uint32_t subfunction, std::uint32_t ex[4]) -> void
             {
                 __cpuid_count(level, subfunction, ex[0], ex[1], ex[2], ex[3]);
             }
 
 #    elif BOOST_COMP_MSVC || defined(BOOST_COMP_MSVC_EMULATED) || defined(__INTEL_COMPILER)
 #        include <intrin.h>
-            inline auto cpuid(std::uint32_t const level, std::uint32_t const subfunction, std::uint32_t ex[4]) -> void
+            inline auto cpuid(std::uint32_t level, std::uint32_t subfunction, std::uint32_t ex[4]) -> void
             {
                 __cpuidex(reinterpret_cast<int*>(ex), level, subfunction);
             }
 #    else
-            inline auto cpuid(std::uint32_t const level, std::uint32_t const subfunction, std::uint32_t ex[4]) -> void
+            inline auto cpuid(std::uint32_t, std::uint32_t, std::uint32_t ex[4]) -> void
             {
                 ex[0] = ex[2] = ex[3] = NO_CPUID;
                 ex[1] = UNKNOWN_COMPILER;
             }
 #    endif
 #else
-            inline auto cpuid(std::uint32_t const level, std::uint32_t const subfunction, std::uint32_t ex[4]) -> void
+            inline auto cpuid(std::uint32_t, std::uint32_t, std::uint32_t ex[4]) -> void
             {
                 ex[0] = ex[2] = ex[3] = NO_CPUID;
                 ex[1] = UNKNOWN_CPU;

